src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape : Shape;
+  color : Color;
+  getArea():number;
 }
 
-export class Triangle {
+abstract class BaseFigure implements Figure {
+  shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
+
+  protected allSide(sides: number[]): void {
+    if (sides.some((value: number) => value <= 0)) {
+      throw new Error(
+        `figure of shape ${this.shape} should have only positive params`,
+      );
+    }
+  }
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  readonly shap = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    super();
+
+    this.allSide([a, b, c]);
+
+    const SortSide: number[] = [a, b, c]
+      .sort((prevSide: number, currentSide: number) => prevSide - currentSide);
 
+    if (SortSide[0] >= SortSide[1] + SortSide[2]) {
+      throw new Error('side is not correct');
+    }
+  }
+
+  getArea(): number {
+    const perimetr = (this.a + this.b + this.c) / 2;
```

Comment:
```suggestion
    const halfPerimeter = (this.a + this.b + this.c) / 2;
```
it is really half perimeter